src/Microsoft.Data.Analysis/PrimitiveDataFrameColumnRollingWindow.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Microsoft.Data.Analysis
+{
+    /// <summary>
+    /// A window for <see cref="PrimitiveDataFrameColumn{T}"/> to support rolling window operations
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public class PrimitiveDataFrameColumnRollingWindow<T> : DataFrameColumnWindow
+        where T : unmanaged
+    {
+        private int _windowSize;
+        private PrimitiveDataFrameColumn<T> _currentColumn;
+        private long _currentIndex;
```

Comment:
Why is `_currentIndex` a field of the class? It feels like it should be a local variable in the methods that use it.

Reply:
Initially I was going to have just 1 `RollingWindow` object and reset the `_currentColumn` and `_currentIndex` as needed. But, since I'm not adding `Rolling` on `DataFrame` yet, we could change `_currentIndex` to a local.